packages/cactus-plugin-ledger-connector-daml/src/main/typescript/plugin-ledger-connector-daml.ts

```diff
@@ -0,0 +1,318 @@
+import type { Server } from "http";
+import type { Server as SecureServer } from "https";
+import type { Config as SshConfig } from "node-ssh";
+import type { Express } from "express";
+// import urlcat from "urlcat";
+import OAS from "../json/openapi.json";
+import axios from 'axios';
+// import { BadRequestError, GatewayTimeoutError } from "http-errors-enhanced-cjs";
+
+import {
+  GetPartiesInvolvedEndpointRequest,
+  QueryIOUEndpointRequest,
+  QueryIOUEndpointResponse,
+  QueryRawIOUEndpointRequest,
+  CreateIOUEndpointRequest,
+  CreateIOUEndpointResponse,
+  ExerciseIOUEndpointRequest,
+  ExerciseIOUEndpointResponse
+} from "./generated/openapi/typescript-axios";
+
+import {
+  IPluginLedgerConnector,
+  IWebServiceEndpoint,
+  IPluginWebService,
+  ICactusPluginOptions,
+  ConsensusAlgorithmFamily,
+  ICactusPlugin
+} from "@hyperledger/cactus-core-api";
+import { consensusHasTransactionFinality } from "@hyperledger/cactus-core";
+import {
+  Checks,
+  Logger,
+  LoggerProvider,
+  LogLevelDesc,
+} from "@hyperledger/cactus-common";
+
+import {
+  ICreateIOUEndpointOptions,
+  CreateIOUEndpoint
+} from "./web-services/create-iou-endpoint";
+
+import {
+  IExerciseIOUEndpointOptions,
+  ExerciseIOUEndpoint
+} from "./web-services/exercise-iou-endpoint";
+
+import {
+  IQueryIOUEndpointOptions,
+  QueryIOUEndpoint
+} from "./web-services/query-iou-endpoint";
+
+import {
+  IQueryRawIOUEndpointOptions,
+  QueryRawIOUEndpoint
+} from "./web-services/query-raw-iou-endpoint";
+
+import {
+  IGetPartiesInvolvedEndpointOptions,
+  GetPartiesInvolvedEndpoint
+} from "./web-services/get-parties-involved-endpoint";
+
+import fs from "fs";
+import fetch from "node-fetch";
+import https from "https";
+import * as crypto from 'crypto';
+
+export interface IPluginLedgerConnectorDAMLOptions
+  extends ICactusPluginOptions {
+  logLevel?: LogLevelDesc;
+  apiUrl: string;
+}
+
+export class PluginLedgerConnectorDAML
+  implements
+  IPluginLedgerConnector<
+    CreateIOUEndpointResponse,
+    CreateIOUEndpointRequest,
+    ExerciseIOUEndpointRequest,
+    ExerciseIOUEndpointResponse
+  >,
+    ICactusPlugin,
+    IPluginWebService
+{
+
+  private readonly instanceId: string;
+  private readonly log: Logger;
+  private endpoints: IWebServiceEndpoint[] | undefined;
+
+  constructor(public readonly options: IPluginLedgerConnectorDAMLOptions) {
+    const level = options.logLevel || "INFO";
+    const label = "plugin-ledger-connector-daml";
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.instanceId = this.options.instanceId;
+  }
+
+  public getPackageName(): string {
+    return "@hyperledger/cactus-plugin-ledger-connector-daml";
+  }
+
+  public async transact(): Promise<any> {
+    return null as any;
```

Comment:
@jagpreetsinghsasan I agree, this looks closer to something application specific instead of belong in the generic framework code. I'm not an expert on DAML/Canton, but maybe we are looking for something like this instead? 
https://docs.daml.com/app-dev/grpc/proto-docs.html#submitandwait-method-version-com-daml-ledger-api-v1